stdlib/public/core/Integers.swift.gyb

```diff
@@ -3247,208 +3294,348 @@ public struct DoubleWidth<T : FixedWidthInteger>
   }
 
   public static var isSigned: Bool {
-    return T.isSigned
+    return Base.isSigned
   }
 
   // fixed width
   //
-  public static var max: DoubleWidth<T> {
+  public static var max: DoubleWidth {
     return self.init((High.max, Low.max))
   }
 
-  public static var min: DoubleWidth<T> {
+  public static var min: DoubleWidth {
     return self.init((High.min, Low.min))
   }
 
-  public static var bitWidth : Int { return 2 * T.bitWidth }
+  public static var bitWidth: Int {
+    return 2 * Base.bitWidth
+  }
 
 % # This covers + and -
 % for x in binaryArithmetic['Numeric'][:2]:
-%   highAffectedByLowOverflow = 'T.max' if x.operator == '+' else 'T.min'
-  public func ${x.name}ReportingOverflow(_ rhs: DoubleWidth<T>)
-    -> (partialValue: DoubleWidth<T>, overflow: ArithmeticOverflow) {
+%   highAffectedByLowOverflow = 'Base.max' if x.operator == '+' else 'Base.min'
+  public func ${x.name}ReportingOverflow(_ rhs: DoubleWidth)
+    -> (partialValue: DoubleWidth, overflow: ArithmeticOverflow) {
     let (low, lowOverflow) =
       _storage.low.${x.name}ReportingOverflow(rhs._storage.low)
     let (high, highOverflow) =
       _storage.high.${x.name}ReportingOverflow(rhs._storage.high)
     let isLowOverflow = lowOverflow == .overflow
-    let result = (high ${x.operator} (isLowOverflow ? 1 : 0), low)
+    let result = (high ${x.operator} (isLowOverflow ? High(1) : High(0)), low)
     let overflow = ArithmeticOverflow(
       highOverflow == .overflow ||
       high == ${highAffectedByLowOverflow} && isLowOverflow
     )
-    return (partialValue: DoubleWidth<T>(result),
+    return (partialValue: DoubleWidth(result),
       overflow: overflow)
   }
 % end
 
-
   public func multipliedReportingOverflow(
-    by rhs: DoubleWidth<T>
-  ) -> (partialValue: DoubleWidth<T>, overflow: ArithmeticOverflow) {
-    let isNegative = (self < DoubleWidth<T>()) != (rhs < DoubleWidth<T>())
-
-    func mul(_ x: Low, _ y: Low, _ carry: Low) -> (partial: Low, carry: Low) {
-      let pair = x.multipliedFullWidth(by: y)
-      let t = DoubleWidth<Low>(pair) + DoubleWidth<Low>((0, carry))
-      return (partial: t._storage.low, carry: t._storage.high)
-    }
-
-    var high: Low = 0
-    var low: Low = 0
-
-    func mkResult(_ isOverflow: Bool)
-      -> (partialValue: DoubleWidth<T>, overflow: ArithmeticOverflow) {
+    by rhs: DoubleWidth
+  ) -> (partialValue: DoubleWidth, overflow: ArithmeticOverflow) {
+    let (carry, product) = multipliedFullWidth(by: rhs)
+    let result = DoubleWidth(product)
+    
+    let isNegative = (self < DoubleWidth()) != (rhs < DoubleWidth())
+    let overflow = isNegative
+      ? carry != ~DoubleWidth()
+      : carry != DoubleWidth()
+    return (result, ArithmeticOverflow(overflow))
+  }
+
+  public func quotientAndRemainder(dividingBy other: DoubleWidth)
+    -> (quotient: DoubleWidth, remainder: DoubleWidth) {
+    let isNegative = (self < DoubleWidth()) != (other < DoubleWidth())
```

Comment:
I think this was from before I implemented the integer literal init—works now for values that aren't larger than an `Int`.